pandas/_libs/index.pyx

```diff
@@ -51,21 +51,9 @@ cpdef get_value_at(ndarray arr, object loc, object tz=None):
 
 cpdef object get_value_box(ndarray arr, object loc):
     cdef:
-        Py_ssize_t i, sz
-
-    if util.is_float_object(loc):
-        casted = int(loc)
-        if casted == loc:
-            loc = casted
-    i = <Py_ssize_t> loc
-    sz = cnp.PyArray_SIZE(arr)
-
-    if i < 0 and sz > 0:
-        i += sz
-
-    if i >= sz or sz == 0 or i < 0:
-        raise IndexError('index out of bounds')
+        Py_ssize_t i
 
+    i = util.validate_indexer(arr, loc)
```

Comment:
Since `get_value_at` (the util version below, which is called by the `get_value_at` in this file) now has the same validation, is it then not unnecessary to call the validation here as well?

Reply:
I think you’re right, will update.